lib/matobjplist.gi

```diff
@@ -18,48 +18,45 @@
 # Constructors:
 ############################################################################
 
-InstallGlobalFunction( MakePlistVectorType,
-  function( basedomain, filter )
-    local T, filter2;
-    filter2 := filter and IsMutable;
+BindGlobal( "MakeIsPlistVectorRep",
+  function(basedomain, mutable, list )
+    local filter, typ;
+    # TODO: cache filter and type in the family obj;
+    # we need variants for mutable vs. immutable, and also for
+    # CanEasilyCompareElements true/false; we might reduce this
+    # by using CanEasilyCompareElementsFamily
```

Comment:
The base domain may not be a component object, so it is not clear how to store something in it in general.

Also, the type itself does not depend on the basedomain at all, so it seems sensibel to share it between different basedomains in the same family?

Reply:
In the current example, the type depends on whether the base domain is `Integers` or not, and one can think about similar situations where special types for vectors/matrices over prime fields are of interest.
We can decide that we do not support this feature.

If we want to support this feature then the "default vector/matrix object types" that should be cached for objects in a given family depend on the constructing filter, the base domain, and the mutability.
Caching these types in the family is possible, but if we can cache them in the base domain then this is simpler. For example, a cache in the family should be "weak" in the sense that entries can be deleted whose base domain parts are not referenced in other places.

Reply:
But the filter and the type do not reference the basedomain at all. So I don't see why we need them to be "weak". Yes, *in theory* we might need something more general, but I'd rather solve that problem then, instead of trying to anticipate the general case now and implementing a solution for that (currently) hypothetical scenario -- in my experience doing that just leads to complexity that is not actually helpful when the real thing happens.

So instead, the caching I now pushed handles integers and finite fields separately (storying the types for those in the domains). We can easily do this *differently* later when the need arises, but for now it seems to work well. I'll provide some benchmark results below.

Reply:
What I meant was that if the family caches some type that depends on the base domain then the *key* for the entry involves the base domain.

Thanks for updates.